x-pack/plugins/fleet/server/services/epm/packages/bundled_packages.ts

```diff
@@ -38,18 +47,20 @@ export async function getBundledPackages(): Promise<BundledPackage[]> {
 
     const result = await Promise.all(
       zipFiles.map(async (zipFile) => {
-        const file = await fs.readFile(path.join(bundledPackageLocation, zipFile));
-
         const { pkgName, pkgVersion } = splitPkgKey(zipFile.replace(/\.zip$/, ''));
 
+        const getBuffer = () => fs.readFile(path.join(bundledPackageLocation, zipFile));
```

Comment:
@dgieselaar with the `once` this will cache the file in memory forever right? it's probably something we want to avoid, we are trying to reduce the fleet memory footprint in kibana too.

Reply:
what we try to achieve with that PR is to avoid to load all the bundled package in memory when we call `getBundledPackage` (that is called in a lot of place in fleet code)

Reply:
@dgieselaar that PR changed the `getBundledPackages` to cache the results, so those bundled package object we are returning are now global, I made a change in https://github.com/elastic/kibana/pull/172640/commits/d2c575ed75a59420b4deb9c2181a326bd1cbc027 to make a copy for each `getBundledPackages` and use a `once` there does it make sense to you?

Reply:
ah, yes, makes sense!